pkg/controller/batchrelease/workloads/commons.go

```diff
@@ -0,0 +1,194 @@
+package workloads
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/openkruise/rollouts/api/v1alpha1"
+	"github.com/openkruise/rollouts/pkg/util"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/klog/v2"
+	"k8s.io/utils/integer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func filterPodsForRollback(pods []*corev1.Pod, plannedBatchCanaryReplicas, expectedBatchStableReplicas, replicas int32, rolloutID, updateRevision string) []*corev1.Pod {
+	var noNeedRollbackReplicas int32
+	var realNeedRollbackReplicas int32
+	var expectedRollbackReplicas int32 // total need rollback
+
+	var terminatingPods []*corev1.Pod
+	var needRollbackPods []*corev1.Pod
+	var noNeedRollbackPods []*corev1.Pod
+
+	for _, pod := range pods {
+		if !pod.DeletionTimestamp.IsZero() {
```

Comment:
They are has some different between filterPodsForRollback and countNoNeedRollbackReplicas about this logic ... 

It may be hard to wrap this blocks ...